### ISSUE TITLE
Reproducability

### ISSUE THREAD

**Issue by mkcifci** [ThilinaRajapakse/simpletransformers#1459]
Hi, 

I have an issue with reproducing results of binary classification using BERT. 

I have seeds set. Why am I getting different results? Why is the result of models not stable? 

def seed_all(seed_value):
    random.seed(seed_value) # Python
    np.random.seed(seed_value) # cpu vars
    torch.manual_seed(seed_value) # cpu  vars
    if torch.cuda.is_available(): 
        torch.cuda.manual_seed(seed_value)
        torch.cuda.manual_seed_all(seed_value) # gpu vars
        torch.backends.cudnn.deterministic = True  #needed
        torch.backends.cudnn.benchmark = False

seed_all(4)

train_args = {
    "output_dir": "outputs/",
    'max_seq_length': 512,
    'num_train_epochs': 5, # 5 
    'train_batch_size': 16,
    'eval_batch_size': 20, # 32
    'gradient_accumulation_steps': 1,
    'learning_rate': 5e-5,
    'warmup_steps': 100,
    "weight_decacy":0.01,
    "logging_steps":100,
    'evaluate_during_training': True,
   'reprocess_input_data': True,
    "save_model_every_epoch": False,
    'overwrite_output_dir': True,
    'no_cache': True,
    'use_early_stopping': True,
    'early_stopping_patience': 3,
    'manual_seed': 4,  
    'use_multiprocessing': False,
    'use_multiprocessing_for_evaluation': False,
}

I define the model as; 
model = ClassificationModel(
    "bert", "bert-base-uncased", args=train_args, use_cuda=True,num_labels=2
    )
